libass/ass_cache.c

```diff
@@ -40,12 +40,13 @@
 // font cache
 static bool font_key_move(void *dst, void *src)
 {
-    ASS_FontDesc *k = src;
-    if (dst)
-        memcpy(dst, src, sizeof(ASS_FontDesc));
-    else
-        free(k->family);
-    return true;
+    if (!dst)
+        return true;
+
+    ASS_FontDesc *d = dst, *s = dst;
+    memcpy(dst, src, sizeof(ASS_FontDesc));
```

Comment:
If `s` were to equal `src`, then this could be simplified to `*d = *s;`

Reply:
I were thinking about that too, but there is a bunch of `memcpy()` in the same context so I decided to use it. Maybe it's indeed better to replace all such `memcpy()` with assignment as that is less error-prone.